### PR TITLE
Use callbag-from-iter under the hood

### PR DESCRIPTION
**Pull request by Andarist**
Let's face it `loop`, `inLoop`, `got1`, `ended` etc dance is quirky, hard to follow. IMHO it's much better to contain it in a single package (when possible) and just reuse it.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,37 +13,19 @@ const checkArgs = (from, to, step) => {
```
Comment (franciscotln) [inline index.js:16]: I've noticed that it's possible to get into an infinite loop if someone enters a descending range with a positive step, so please add one more check inside the 'checkArgs' function
```js
if (from > to && step > 0) {
  throw new Error('Descending range must have negative step. Got +' + step);
}
```

File: index.js [hunk 2]
```diff
@@ -11,37 +13,19 @@ const checkArgs = (from, to, step) => {
...
-        if ((to > from && sent <= to) || (to < from && sent >= to)) {
-          sink(1, sent);
-        } else {
-          ended = true;
-          sink(2);
-        }
-      }
-      inLoop = false;
-    };
+  let sent = from - step;
```
Comment (franciscotln) [inline index.js]: you can name this variable as `value` instead of `sent`

File: package.json [hunk 3]
```diff
@@ -1,7 +1,9 @@
 {
   "author": "Francisco Tavares de Lima Neto",
   "description": "Callbag source that generates a range of numbers",
-  "dependencies": {},
+  "dependencies": {
+    "callbag-from-iter": "^1.2.0"
```
Comment (franciscotln) [inline package.json:5]: please use minor versioning `"~1.2.0"`

**Review by franciscotln — CHANGES_REQUESTED** [review]

**Comment by franciscotln** [comment]
@Andarist there are also some conflicts in index.js :-)

File: index.js (hunk 1 above)
Comment (franciscotln) [reply index.js:16]: this should be at the end of the checkArgs function to be sure that all args are values :-)

**Comment by Andarist** [comment]
I've fixed the conflict & adjusted the code according to your requests (mostly).

I've also added some additional stuff (latest 2 commits) as part of this PR because I didnt want to have guaranteed conflicts later. Let me know what do you think.

File: package.json (hunk 3 above)
Comment (Andarist) [reply package.json:5]: What's the reasoning behind this? It's rather not a common practice and might cause duplicated dependencies in people's applications.

File: index.js [hunk 4]
```diff
@@ -7,42 +9,31 @@ const checkArgs = (from, to, step) => {
   if (!isNumber(from) || !isNumber(to) || !isNumber(step)) {
     throw new Error('Arguments must be numbers');
   }
+  if (from > to && step > 0) {
+    throw new Error('Descending range must have negative step. Got +' + step);
+  }
 };
 
+const isProd = process.env.NODE_ENV === 'production'
```
Comment (Andarist) [inline index.js:17]: this should allow dropping the checkArgs in production mode when bundling this library for usage on the web

Comment (franciscotln) [reply index.js:17]: yeah, good point

File: package.json (hunk 3 above)
Comment (franciscotln) [reply package.json:5]: Ok, I agree with you. On one hand I don't want to have dependencies breaking the behaviour of our implementation, on the other hand it's not good to duplicate dependencies. You can leave it as is.

Comment (Andarist) [reply package.json:5]: minor versions shouldnt break the backward compatibility, so it should be "safe" to use ^ instead of ~, but ofc they might because semver versions are bumped by humans (although the same argument applies to ~ - somebody might release a breaking change in patch version)
